### ISSUE TITLE
feat: support downloading github binaries from private repo

### ISSUE THREAD

**Issue by BubbaJoe** [cloudquery/cloudquery#11409]
### Which problem is this feature request solving?

Hello, I am trying to download my release, but it's in a private repo, which means the release is also private. So it seems it isnt possible to use a github token or something to allow this?

### Describe the solution you'd like

```
registry: github
github_access_token: ${GITHUB_TOKEN}
```

## helpful links

https://github.com/cloudquery/cloudquery/blob/main/cli/internal/plugin/managedsource/plugin.go#L133
https://github.com/cloudquery/plugin-pb-go/blob/main/specs/source.go
https://stackoverflow.com/questions/20396329/how-to-download-github-release-from-private-repo-using-command-line

### Pull request (optional)

- [X] I can submit a pull request

**Comment by yevgenypats** [comment]
Commenting in the meanwhile that a possible workaround is to download the private release via `wget` as a "pre" step and then use `local` for registry and point to the downloaded plugin.

**Comment by BubbaJoe** [comment]
@yevgenypats Concerning the design, can i just use environment variables and not require the value to be added in the configuration. The amount of work required to do this is a lot.

**Comment by yevgenypats** [comment]
We introduced CloudQuery Hub (registry) recently - https://hub.cloudquery.io/ . This will be the official registry we will support going forward as we can't support all registries. But this should open up the possibility to publish to CloudQuery Registry from GitLab/GitHub or any other CI/CD. It's still possible to host CQ plugin anywhere else and download via `wget` so cloudquery cli can run the downloaded plugin locally.

CloudQuery Hub is also supporting publishing and downloading private plugins.

Closing this as complete.
